src/pynxtools_xps/reader.py

```diff
@@ -72,439 +72,575 @@
+        if not isinstance(value2, list):
+            value2 = [value2]
+        concatenated = value1 + value2
 
-            template[f"{modified_key}_errors"] = np.std(
-                [
-                    xr_data[x_arr].data
-                    for x_arr in xr_data.data_vars
-                    if (chan_count not in x_arr and scan_count in x_arr)
-                ],
-                axis=0,
-            )
-            template[f"{modified_key}_errors/@units"] = units
+    return concatenated
 
-            template[f"{modified_key}/@long_name"] = "XPS intensity"
 
+# pylint: disable=too-few-public-methods
+class XPSReader(MultiFormatReader):
+    """Reader for XPS."""
 
-def fill_detector_group(key, entries_values, config_dict, xps_data_dict, template):
-    """Fill out fields and attributes for NXdetector/NXdata"""
+    supported_nxdls = [
+        "NXmpes",
+        # "NXxps",
+    ]
 
-    chan_count = "_chan"
-    scan_count = "_scan"
+    reader_dir = Path(__file__).parent
+    config_file: Optional[Union[str, Path]] = reader_dir.joinpath(
+        "config", "template.json"
+    )
+
+    __prmt_file_ext__ = [
+        ".ibw",
+        ".npl",
+        ".pro",
+        ".spe",
+        ".sle",
+        ".slh",
+        ".txt",
+        ".vms",
+        ".xml",
+        ".xy",
+    ]
+    __vendors__ = ["kratos", "phi", "scienta", "specs", "unkwown"]
+    __prmt_vndr_cls: Dict[str, Dict] = {
+        ".ibw": {"scienta": MapperScienta},
+        ".npl": {"unkwown": VamasMapper},
+        ".pro": {"phi": MapperPhi},
+        ".spe": {"phi": MapperPhi},
+        ".sle": {"specs": SleMapperSpecs},
+        ".txt": {
+            "scienta": MapperScienta,
+            "unknown": TxtMapperVamasExport,
+        },
+        ".vms": {"unkwown": VamasMapper},
+        ".xml": {"specs": XmlMapperSpecs},
+        ".xy": {"specs": XyMapperSpecs},
+    }
+
+    __file_err_msg__ = (
+        "Need an XPS data file with the following extension: " f"{__prmt_file_ext__}"
+    )
+
+    __vndr_err_msg__ = (
+        "Need an XPS data file from the following vendors: " f"{__vendors__}"
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.xps_data: Dict[str, Any] = {}
+        self.eln_data: Dict[str, Any] = {}
+
+        self.extensions = {
+            ".yml": self.handle_eln_file,
+            ".yaml": self.handle_eln_file,
+            ".json": self.set_config_file,
+        }
 
-    unit_config_value = config_dict[f"{key}/@units"]
+        for ext in XPSReader.__prmt_file_ext__:
+            self.extensions[ext] = self.handle_data_file
 
-    for entry, xr_data in entries_values.items():
-        modified_key = key.replace("/ENTRY[entry]/", f"/ENTRY[{entry}]/")
+        self.callbacks = ParseJsonCallbacks(
+            attrs_callback=self.get_attr,
+            data_callback=self.get_data,
+            eln_callback=self.get_eln_data,
+            dims=self.get_data_dims,
+        )
```

Comment:
```suggestion
```
You don't need to set this here as long as you call `super().__init__(...)` and use the default names (which you both do).

Reply:
Yeah, I had them in when I was debugging. Removed now.